kv/task.go

```diff
@@ -425,21 +404,15 @@ func (s *Service) findTasksByOrg(ctx context.Context, tx Tx, filter influxdb.Tas
 			break
 		}
 
-		if !taskFilterMatch(filter.Type, t.Type) {
-			continue
-		}
-
-		// insert the new task into the list
-		ts = append(ts, t)
-
-		// Check if we are over running the limit
-		if len(ts) >= filter.Limit {
-			break
+		if matchFn == nil || matchFn(t) {
+			ts = append(ts, t)
+			// Check if we are over running the limit
+			if len(ts) >= filter.Limit {
+				break
+			}
 		}
-	}
 
-	if filter.Name != nil {
-		ts = filterByName(ts, *filter.Name)
+		k, v = c.Next()
 	}
```

Comment:
Great! 

This fixes the same bug that I discovered in `findAllTasks`, where the limit is applied before we complete all filtering. It is possible for all tasks to be filtered out by `Name`, preventing the caller from moving to the next page. By moving the check to the `matchFn`, we ensure that the limit is applied after filtering has happened.